src/codegen/tests/snapshots/rusty__codegen__tests__code_gen_tests__reference_to_reference_assignments_in_function_arguments.snap

```diff
@@ -0,0 +1,42 @@
+---
+source: src/codegen/tests/code_gen_tests.rs
+expression: result
+---
+; ModuleID = 'main'
+source_filename = "main"
+
+%STRUCT_params = type { i8, i8, i8 }
+%prog = type { %STRUCT_params*, %STRUCT_params*, %STRUCT_params* }
+%main = type {}
+
+@global1 = global %STRUCT_params zeroinitializer
+@global2 = global %STRUCT_params zeroinitializer
+@global3 = global %STRUCT_params zeroinitializer
+@global4 = global i32 0
+@global5 = global [81 x i8] zeroinitializer
+@global6 = global float 0.000000e+00
+@prog_instance = global %prog zeroinitializer
+@main_instance = global %main zeroinitializer
+@__STRUCT_params__init = unnamed_addr constant %STRUCT_params zeroinitializer
+
+define void @prog(%prog* %0) {
+entry:
+  %input1 = getelementptr inbounds %prog, %prog* %0, i32 0, i32 0
+  %input2 = getelementptr inbounds %prog, %prog* %0, i32 0, i32 1
+  %input3 = getelementptr inbounds %prog, %prog* %0, i32 0, i32 2
+  ret void
+}
+
+define void @main(%main* %0) {
+entry:
+  store %STRUCT_params* @global1, %STRUCT_params** getelementptr inbounds (%prog, %prog* @prog_instance, i32 0, i32 0), align 8
+  store %STRUCT_params* @global2, %STRUCT_params** getelementptr inbounds (%prog, %prog* @prog_instance, i32 0, i32 1), align 8
+  store %STRUCT_params* @global3, %STRUCT_params** getelementptr inbounds (%prog, %prog* @prog_instance, i32 0, i32 2), align 8
+  call void @prog(%prog* @prog_instance)
+  store %STRUCT_params* bitcast (i32* @global4 to %STRUCT_params*), %STRUCT_params** getelementptr inbounds (%prog, %prog* @prog_instance, i32 0, i32 0), align 8
+  store %STRUCT_params* bitcast ([81 x i8]* @global5 to %STRUCT_params*), %STRUCT_params** getelementptr inbounds (%prog, %prog* @prog_instance, i32 0, i32 1), align 8
```

Comment:
I'm not sure about this line, any reason why it's `[81 x i8]`?

Reply:
it's the the default length of the type STRING
you can also define the size explicitly -> STRING[10]